v2/pkg/engine/plan/visitor.go

```diff
@@ -707,10 +707,22 @@ func (v *Visitor) resolveFieldValue(fieldRef, typeRef int, nullable bool, path [
 				}
 			}
 		case ast.NodeKindEnumTypeDefinition:
-			return &resolve.String{
-				Path:                 path,
-				Nullable:             nullable,
-				UnescapeResponseJson: unescapeResponseJson,
+			values := make([]string, 0, len(v.Definition.EnumTypeDefinitions[typeDefinitionNode.Ref].EnumValuesDefinition.Refs))
+			inaccessibleValues := make([]string, 0)
+			for _, valueRef := range v.Definition.EnumTypeDefinitions[typeDefinitionNode.Ref].EnumValuesDefinition.Refs {
+				valueName := v.Definition.EnumValueDefinitionNameString(valueRef)
+				values = append(values, valueName)
+
+				if _, isInaccessible := v.Definition.EnumValueDefinitionDirectiveByName(valueRef, []byte("inaccessible")); isInaccessible {
```

Comment:
I think it would be better to add inaccessible as a feature to our AST and not have it as implicit property of a graph

Reply:
It is not ast feature, it is schema behavior 

didn't like the idea to have it directly in the ast